### DISCUSSION TITLE
Creating login pages via React (headless auth)

### DISCUSSION THREAD

**Discussion by Nefcanto** [keycloak/keycloak#35922] [Ideas]
### Description

It would be great if we could create login pages using technologies that we are comfortable with.
I say the login section (login, forgot password, etc.) because other parts can already be developed using REST API.
For example, we have a React panel in which we let users manage their data, or let admins create users and assign roles with Keycloak API as the backend.

It would be an amazing DX if we could get headless auth.

### Motivation

Keycloak is made of some parts.

- Login
- User account
- Admin panel
- CLI

Teams need to customize or integrate these parts into their applications. Login can be customized, but it's hard for teams not familiar with Java, Patternfly, Freemaker template, etc.

They can integrate user account and admin features into their existing applications, using Keycloak REST API.
But the login page is tightly coupled.

### Details

We can use `keycloak` JavaScript client library. It can expose some utility methods to check some known attack possibilities (for example open redirection). It can act as a headless auth provider/checker.

**Comment by AlexDev404** [comment]
Any updates on this? @stianst
